### PR TITLE
`azurerm_private_endpoint_connection` - Export `network_interface` attributes from private endpoints

### LINKED ISSUE #12283: Support for output of network interface id or name created by private endpoint
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Creating a Private Endpoint will implicitly create a Network Interface, which is mostly managed by the Private Endpoint and shares its Lifecycle. Network Interfaces allow configuration of Diagnostic Settings to fetch metrics, which can be configured within the Portal, but there seems to be currently no way to access the Network Interface that is created by the Private Endpoint in order to configure its Diagnostic Settings, making it impossible to get any metrics. Outputting the id or name of this Network Interface should alleviate the problem, making it possible to use it as input for the azurerm_monitor_diagnostic_setting resource.

### New or Affected Resource(s)

azurerm_private_endpoint

### PR DESCRIPTION
**Pull request by owenfarrell**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### Summary

This pull request exports the ID and name of the network interface associated with private endpoints.

* The `TestAccPrivateEndpoint_privateConnectionAlias` test seems to be a bit flaky. I haven't confirmed the root cause as to why this test intermittently fails, but I have confirmed that it's not the result of these changes.

Fixes: #12283

Output from acceptance testing:

```
$ make acctests SERVICE='network' TESTARGS='-run="TestAcc(PrivateEndpointConnection|PrivateEndpoint)_"' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/network -run="TestAcc(PrivateEndpointConnection|PrivateEndpoint)_" -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccPrivateEndpoint_basic
=== PAUSE TestAccPrivateEndpoint_basic
=== RUN   TestAccPrivateEndpoint_updateTag
=== PAUSE TestAccPrivateEndpoint_updateTag
=== RUN   TestAccPrivateEndpoint_requestMessage
=== PAUSE TestAccPrivateEndpoint_requestMessage
=== RUN   TestAccPrivateEndpoint_privateDnsZoneGroup
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneGroup
=== RUN   TestAccPrivateEndpoint_privateDnsZoneRename
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneRename
=== RUN   TestAccPrivateEndpoint_privateDnsZoneUpdate
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneUpdate
=== RUN   TestAccPrivateEndpoint_privateDnsZoneRemove
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneRemove
=== RUN   TestAccPrivateEndpoint_privateConnectionAlias
=== PAUSE TestAccPrivateEndpoint_privateConnectionAlias
=== CONT  TestAccPrivateEndpoint_basic
=== CONT  TestAccPrivateEndpoint_privateDnsZoneRename
=== CONT  TestAccPrivateEndpoint_requestMessage
=== CONT  TestAccPrivateEndpoint_privateDnsZoneGroup
=== CONT  TestAccPrivateEndpoint_updateTag
=== CONT  TestAccPrivateEndpoint_privateDnsZoneRemove
=== CONT  TestAccPrivateEndpoint_privateConnectionAlias
=== CONT  TestAccPrivateEndpoint_privateDnsZoneUpdate
--- PASS: TestAccPrivateEndpoint_basic (301.98s)
--- PASS: TestAccPrivateEndpoint_privateConnectionAlias (329.88s)
--- PASS: TestAccPrivateEndpoint_requestMessage (379.03s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneGroup (457.55s)
--- PASS: TestAccPrivateEndpoint_updateTag (543.34s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneRename (563.73s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneRemove (643.23s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneUpdate (1228.28s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       1230.978s
```

### CODE REVIEW

File: website/docs/r/private_endpoint.html.markdown [hunk 1]
```diff
@@ -226,6 +234,8 @@ A `private_service_connection` block exports:
 
 * `private_ip_address` - (Computed) The private IP address associated with the private endpoint, note that you will have a private IP address assigned to the private endpoint even if the connection request was `Rejected`.
 
+* `network_interface_id` - (Computed) The private IP address associated with the private endpoint, note that you will have a private IP address assigned to the private endpoint even if the connection request was `Rejected`.
```
Comment (mbfrahry) [inline website/docs/r/private_endpoint.html.markdown:237]: I can't find this attribute listed in the `private_service_connection` block. Did we want to get that added or should we remove this line from the docs?

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @owenfarrell! This PR looks good but I just had one question for the docs. We should be able to get this merged once that is taken care of

File: website/docs/r/private_endpoint.html.markdown (hunk 1 above)
Comment (owenfarrell) [reply website/docs/r/private_endpoint.html.markdown]: @mbfrahry - Good call, that was some leftover copy pasta as I was trying to figure out the "right" way/place to export these attributes. I've amended the last commit to remove that from the docs and keep everything clean(er).

**Comment by owenfarrell** [comment]
@mbfrahry - Thanks for the review. That documentation is now cleaned up.

I also noticed that my test case regex missed the data source test case. Below is an updated acceptance test execution which includes that last (9th) test case.

```
make acctests SERVICE='network' TESTARGS='-run="TestAcc(DataSourcePrivateEndpointConnection|PrivateEndpoint)_"' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/network -run="TestAcc(DataSourcePrivateEndpointConnection|PrivateEndpoint)_" -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourcePrivateEndpointConnection_complete
=== PAUSE TestAccDataSourcePrivateEndpointConnection_complete
=== RUN   TestAccPrivateEndpoint_basic
=== PAUSE TestAccPrivateEndpoint_basic
=== RUN   TestAccPrivateEndpoint_updateTag
=== PAUSE TestAccPrivateEndpoint_updateTag
=== RUN   TestAccPrivateEndpoint_requestMessage
=== PAUSE TestAccPrivateEndpoint_requestMessage
=== RUN   TestAccPrivateEndpoint_privateDnsZoneGroup
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneGroup
=== RUN   TestAccPrivateEndpoint_privateDnsZoneRename
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneRename
=== RUN   TestAccPrivateEndpoint_privateDnsZoneUpdate
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneUpdate
=== RUN   TestAccPrivateEndpoint_privateDnsZoneRemove
=== PAUSE TestAccPrivateEndpoint_privateDnsZoneRemove
=== RUN   TestAccPrivateEndpoint_privateConnectionAlias
=== PAUSE TestAccPrivateEndpoint_privateConnectionAlias
=== CONT  TestAccDataSourcePrivateEndpointConnection_complete
=== CONT  TestAccPrivateEndpoint_privateDnsZoneRename
=== CONT  TestAccPrivateEndpoint_requestMessage
=== CONT  TestAccPrivateEndpoint_updateTag
=== CONT  TestAccPrivateEndpoint_basic
=== CONT  TestAccPrivateEndpoint_privateDnsZoneGroup
=== CONT  TestAccPrivateEndpoint_privateDnsZoneRemove
=== CONT  TestAccPrivateEndpoint_privateConnectionAlias
--- PASS: TestAccPrivateEndpoint_privateConnectionAlias (288.15s)
=== CONT  TestAccPrivateEndpoint_privateDnsZoneUpdate
--- PASS: TestAccDataSourcePrivateEndpointConnection_complete (290.60s)
--- PASS: TestAccPrivateEndpoint_basic (362.14s)
--- PASS: TestAccPrivateEndpoint_requestMessage (457.61s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneGroup (465.04s)
--- PASS: TestAccPrivateEndpoint_updateTag (549.46s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneRename (585.16s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneRemove (606.79s)
--- PASS: TestAccPrivateEndpoint_privateDnsZoneUpdate (642.65s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/network       935.009s
```

**Review by mbfrahry — APPROVED** [review]
LGTM! Thanks for getting that doc fixed!

**Pull request merged**